library/src/main/java/io/ona/kujaku/interfaces/IKujakuMapView.java

```diff
@@ -70,11 +70,22 @@
      * intervention. If the MY LOCATION BUTTON is visible, it will turn blue as long as this mode is on.
      * This can be turned off by the user if s/he touches the map to scroll it to a specific location.
      *
-     * @param focusOnMyLocation
+     * @param focusOnMyLocation Whether to focus on the user's current location or not
      */
     void focusOnUserLocation(boolean focusOnMyLocation);
```

Comment:
I tried testing out the other activities with this functionality, it seems that the functionality has broken and I have to click on the **MY LOCATION BUTTON** multiple times(like 10-20) for the map to zoom to my location as it did previously. 
1. Is this the expected functionality or has is it a bug?
2. If this is the expected functionality, why change this to the current implementation ? In my humble opinion, it seems a bit more difficult and takes more effort for a user to actually get to see their current location

Reply:
You're right, it's harder now.

It's not a bug, just a CameraMode setting for the LocationComponent.

Easy fix.

Reply:
Is this configurable or is this the default for the feature currently? If it's not configurable and you  need it to be, could you do that and have the default as it was before.

Reply:
One can always access the LocationComponent and reset this, I have removed this behavior as the default setting.

Reply:
1. It's a bit jerky. I don't know why?
2. The buffer radius and it’s blue appearance(of the location component) disappears after some time, is this going to be a problem for the specific use cases of Reveal and other host apps
![screenshot_20190220-115336](https://user-images.githubusercontent.com/31766075/53078946-61e53d00-3506-11e9-8d99-0fa475174726.png)

Reply:
@githengi  could you test this part and provide feedback?

Reply:
It's jerky because the `centerMap` function has a default 1000ms before settling, if I reduced it to 50ms, it is okay.

Any reason it was set to 1000ms?

Reply:
That's the animation duration. Which works OK in the master branch? It stutters while trying to change target location and zoom which if it takes too long(I assume it just did not zoom completely and re-click the button)

Reply:
I’m trying to find out if the duration can be reduced

it’s the one causing the jerky movement

Reply:
The transition to the user location takes time sometimes. Also the transition is not smooth.

However on clicking the increase or decrease radius the response is instant and transition it better

Reply:
I have changed the transition time to 50ms.

I haven't find any issues with this so far.

Reply:
This is not a great experience, where the user location just pops up. I think you should return the transition to 1 second.

Reply:
`setWarmGps(true);` in the onClick method of `currentLocationBtn` in `KujakuMapView` is what is causing jerky updates.

When disabled, the 1 second transitions are smooth.

I really can't tell what it does right now but it is causing some kind of loop that keeps updating locations while the animation is happening.

@ekigamba could you shed some light on what it does and maybe suggest a possible fix?

@githengi @ekigamba I again comment that `KujakuMapView` is growing too big to debug effectively and may need to be broken up.

Reply:
@vincent-karuri I concur, The refactor should be prioritized, probably on the roadmap for the June release. @craigappl

Reply:
@vincent-karuri @Kigamba maybe this is due to this call that is called on `setWarmGps(true)` `io.ona.kujaku.utils.LocationSettingsHelper#checkLocationEnabled`

Its making a location request independently

Reply:
1. It was and is working OK on master (Where there are renders(similar as now) to create the same circle(s))
2. I have checked the code and there is no loop unless you can point it out
3. The issue has come up after adding this code so can you be certain that you cannot fix within the added code

Reply:
I don't think that's not the point.

I was asking what `setWarmGps` does and why it is called when the `currentLocationBtn` is clicked.

I know it works in master and `setWarmGps` calls this function `checkLocationSettingsAndStartLocationServices` which has so much going on (can't comprehend it).

The `checkLocationSettingsAndStartLocationServices` function is convoluted and not easy to understand, hence why I'm asking.

The point is `setWarmGps` is what is not playing nice with `LocationComponent` and that's why I'm asking you to provide more information on what it does.

Reply:
@vincent-karuri 

1. You moved this discussion to Slack yesterday and are yet to respond to the responses I provided let alone try out the solution that I provided since it seems the new code introduced is heavy on rendering. 
2. You went ahead and removed the transition that helps with the UX and ignored it - I hoped that this solution would help at least

Could you work sequentially on the actual problem? And try take the solutions provided. Otherwise, you are taking us in a circle if you can't specify some of these many issues(which I have checked) and do not want to try out the proposed solution. 

I honestly don't think it's fair or worth it to continue any argument and left it at a civil point

Reply:
FYI: I tried your solution out today. Didn't work. That's not the problem.

 I later discovered it had to do with the call to `setWarmGps` after doing my own debugging which took a lot of time as well.

That's when I sought to get your opinion on what might be happening in this function that could be a problem (I believe you were the core implementer).

This is allowed and even encouraged in software engineering. I'm yet to get a response trying to walk me through what may be going on in that function or the functions it calls.

I also specified exactly what function I'm unable to follow. I fail to see what the issue is here.

Reply:
> @vincent-karuri @Kigamba maybe this is due to this call that is called on `setWarmGps(true)` `io.ona.kujaku.utils.LocationSettingsHelper#checkLocationEnabled`
> 
> Its making a location request independently

Thanks for this input Sam. It was invaluable in tracking down the issue.

Reply:
@githengi @Kigamba take a look at the current implementation that I believe addresses the camera focus issue.

Reply:
> 1. It's a bit jerky. I don't know why?
> 2. The buffer radius and it’s blue appearance(of the location component) disappears after some time, is this going to be a problem for the specific use cases of Reveal and other host apps
>    ![screenshot_20190220-115336](https://user-images.githubusercontent.com/31766075/53078946-61e53d00-3506-11e9-8d99-0fa475174726.png)

The grey icon appears in a [stale state](https://github.com/mapbox/mapbox-android-demo/issues/735), clicking on the `current location button` fixes it.

I believe it is important to keep it, thoughts @githengi?

Reply:
@vincent-karuri Great work. 
The location widget now transitions to users location with animation as expected.

Did you figure why after sometime the my location widget turns gray

Reply:
@vincent-karuri we can keep it, however the 30 seconds default to turn my location to gray maybe too small. Change Default to maybe 2 minutes.

Reply:
It would be nice if its configurable just like on mapbox, the host app can decide to disable it or provide the duration when the location becomes stale.

Reply:
All this is configurable via xml and `LocationComponentOptions`.